plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcConnectorTest.java

```diff
@@ -45,4 +54,34 @@ protected Session withSmallRowGroups(Session session)
                 .setCatalogSessionProperty("iceberg", "orc_writer_max_stripe_rows", "10")
                 .build();
     }
+
+    @Test
+    public void testTinyintType()
+            throws Exception
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_tinyint", "(\"_col0\" int)")) {
+            // Insert dummy rows and replace the ORC file that the type is struct<_col0:smallint>
+            // Avoid inserting rows using TestTable to generate only 1 file
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (-127), (NULL)", 2);
+            Path orcFilePath = Path.of((String) computeScalar(format("SELECT DISTINCT file_path FROM \"%s$files\"", table.getName())));
+            Files.copy(new File(getResource("single-tinyint-column.orc").toURI()).toPath(), orcFilePath, REPLACE_EXISTING);
+            Files.delete(orcFilePath.resolveSibling(format(".%s.crc", orcFilePath.getFileName())));
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES 127, NULL");
```

Comment:
-127 lost the sign?

Reply:
It's not lost. I used different values between old and new ORC files to clarify that it's exactly replaced. Changed to use same values since it looked not intuitive.